packages/expo-doctor/src/checks/GlobalPrereqsVersionCheck.ts

```diff
@@ -0,0 +1,113 @@
+import spawnAsync from '@expo/spawn-async';
+import { execSync } from 'child_process';
+import semver from 'semver';
+
+import { DoctorCheck, DoctorCheckParams, DoctorCheckResult } from './checks.types';
+
+const MIN_WATCHMAN_VERSION = '4.6.0';
+const MIN_NPM_VERSION = '3.0.0';
+const CORRECT_NPM_VERSION = 'latest';
+const WARN_NPM_VERSION_RANGES = ['>= 5.0.0 < 5.7.0'];
+const BAD_NPM_VERSION_RANGES = ['>= 5.0.0 <= 5.0.3'];
+
+function _isNpmVersionWithinRanges(npmVersion: string, ranges: string[]) {
+  return ranges.some(range => semver.satisfies(npmVersion, range));
+}
+
+async function _checkNpmVersionAsync(): Promise<string | null> {
+  try {
+    try {
+      const yarnVersionResponse = await spawnAsync('yarnpkg', ['--version']);
+      if (yarnVersionResponse.status === 0) {
+        return null;
```

Comment:
Why do we bail out of the _npm_ version check when yarn is found? All of our checks are hardcoded to use npm now (npm explain etc.), so this seems wrong.